### DISCUSSION TITLE
What is the purpose of `verify` `verify-update` commands?

### DISCUSSION THREAD

**Discussion by sim31** [fwupd/fwupd#5796] [General]
**Describe the question**
So I updated coreboot on my device to 8.40, using gnome-firmware. I saw a verify button there, so I decided to check it. The result was that I got the message that basically said that hashes did not match. But also the option to store the new hash (of the current firmware I suspect). Same thing if I run `fwupdmgr verify` on command line.

So what is the hash that it is checking against? Shouldn't it get updated automatically on an update? Or is something wrong here with the update I just performed?

**fwupd version information**
1.8.10

**Comment by hughsie** [comment]
Verify is "compare what the device says with what the vendor says, falling back to the last known stored state. Verify update stores the last known stored state.
